### PR TITLE
fix: update auth logic

### PR DESCRIPTION
**Pull request by rishi-raj-jain**

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
As auth can be undefined on the server as well, it's absolute necessary to check if auth object exists on the server. Better detection of a client would be of typeof window of typeof document.

File: pages/how/auth.mdx [hunk 1]
```diff
@@ -82,11 +82,13 @@ const mutators = {
       {nodeID, content}: {nodeID: string, content: string}) {
     const node = await getNode(tx, nodeID);
     const auth = tx.auth as MyAuthData;
-    if (auth === undefined || // auth is always undefined on client
```
Comment (aboodman) [inline pages/how/auth.mdx:85]: What was wrong with the prev, was it a compile error or something? I think the documentation should still show some form of checking the presence of `auth`, not `window`.

File: pages/how/auth.mdx [hunk 2]
```diff
@@ -82,11 +82,13 @@ const mutators = {
...
-        auth.access === 'write') {
+    if (typeof window !== 'undefined') {
+     // `auth` is always undefined on client
+     // Allow the mutation
+    }
+    // On the server, check for the user access level
+    if (auth.access === 'write') {
```
Comment (aboodman) [inline pages/how/auth.mdx:90]: I think a better check would be something like:

```ts
// tx.auth is undefined on client. On server it should always be defined.
const auth = tx.auth as MyAuthData | undefined;
if (tx.location === 'client' || auth!.access === 'write') {
  // allow mutation
} else {
  console.error("Access denied");
}
```

Comment (rishi-raj-jain) [reply pages/how/auth.mdx:90]: Pushing the changes then.

**Pull request closed without merging**